jaeger-spark-dependencies-elasticsearch/src/main/java/io/jaegertracing/spark/dependencies/elastic/ElasticsearchDependenciesJob.java

```diff
@@ -0,0 +1,242 @@
+    public Builder day(long day) {
+      this.day = Utils.midnightUTC(day);
+      return this;
+    }
+
+    public ElasticsearchDependenciesJob build() {
+      return new ElasticsearchDependenciesJob(this);
+    }
+  }
+
+  private static String getSystemPropertyAsFileResource(String key) {
+    String prop = System.getProperty(key, "");
+    return prop != null && !prop.isEmpty() ? "file:" + prop : prop;
+  }
+
+  private final String index;
+  private final long day;
+  private final String dateStamp;
+  private final SparkConf conf;
+
+  ElasticsearchDependenciesJob(Builder builder) {
+    this.index = builder.index;
+    this.day = builder.day;
+    SimpleDateFormat df = new SimpleDateFormat("yyyy-MM-dd");
+    df.setTimeZone(TimeZone.getTimeZone("UTC"));
+    this.dateStamp = df.format(new Date(builder.day));
+    this.conf = new SparkConf(true).setMaster(builder.sparkMaster).setAppName(getClass().getName());
+    if (builder.jars != null) {
+      conf.setJars(builder.jars);
+    }
+    if (builder.username != null) {
+      conf.set("es.net.http.auth.user", builder.username);
+    }
+    if (builder.password != null) {
+      conf.set("es.net.http.auth.pass", builder.password);
+    }
+    conf.set("es.nodes", parseHosts(builder.hosts));
+    if (builder.hosts.indexOf("https") != -1) {
+      conf.set("es.net.ssl", "true");
+    }
+    for (Map.Entry<String, String> entry : builder.sparkProperties.entrySet()) {
+      conf.set(entry.getKey(), entry.getValue());
+    }
+  }
+
+  public void run() {
+    run(index + "-span-" + dateStamp,"jaeger-dependencies-" + dateStamp + "/dependencies");
+    log.info("Done");
+  }
+
+  void run(String spanResource, String depResource) {
+    JavaSparkContext sc = new JavaSparkContext(conf);
+    try {
+      JavaPairRDD<String, Iterable<Span>> traces = JavaEsSpark.esJsonRDD(sc, spanResource)
+          .map(new ElasticTupleToSpan())
+          .groupBy(Span::getTraceId);
+
+      traces.foreach(stringIterableTuple2 -> {
+        System.out.println(stringIterableTuple2._1() +"  ->  " +  Lists.newArrayList(stringIterableTuple2._2()).size());
+      });
+
+      List<Dependency> dependencyLinks = DepenencyLinksSparkJob.derive(traces);
+      store(sc, dependencyLinks, depResource);
+    } finally {
+      sc.stop();
+    }
+  }
+
+  private void store(JavaSparkContext javaSparkContext, List<Dependency> dependencyLinks, String resource) {
+    if (dependencyLinks.isEmpty()) {
+      return;
+    }
+
+    String json;
+    try {
+      ObjectMapper objectMapper = new ObjectMapper();
+      json = objectMapper.writeValueAsString(new ElasticsearchDependencies(dependencyLinks, day));
+    } catch (JsonProcessingException e) {
+      throw new IllegalStateException("Could not serialize dependencies", e);
+    }
+
+    JavaEsSpark.saveJsonToEs(javaSparkContext.parallelize(Collections.singletonList(json)), resource);
+  }
+
+  static String parseHosts(String hosts) {
```

Comment:
Could you share the motivation for this ? Looks like the only thing it's doing is add the port to the host, in case it's missing. Am I right? In any case, examples of the input and output would be sufficient.

Reply:
It's taken from zipkin. It's a little bit magical, I am wondering if we should to keep it.

Reply:
Remove it and see if it breaks :) Code that does nothing shouldn't exist.